staging/src/k8s.io/apiserver/pkg/admission/metrics/metrics.go

```diff
@@ -122,7 +122,7 @@ func newAdmissionMetrics() *AdmissionMetrics {
 
 	// Admission webhook metrics. Each webhook is identified by name.
 	webhook := newMetricSet("webhook",
-		[]string{"name", "type", "operation", "rejected"},
+		[]string{"name", "configuration_name", "type", "operation", "rejected"},
```

Comment:
It does not seem like compatible. Do we have a strong argument for changing this?

Reply:
For the compatibility, does that mean will create some new metric item?

Reply:
Adding labels breaks ingestion pipelines which is backwards incompatible. Possibly we should just consider appending configuration name to the webhook name since that preserves backwards compatibility. 

What's the expected cardinality like for configuration name?

Reply:
@logicalhan 
configuration name: `configuration-admission` (should be lower case alphanumeric characters, -, and .)
webhook name: `webhook.example.com`

If we appending configuration name to the webhook name, like this `configuration-admission-webhook.example.com`?

Reply:
What I meant was, how many different configuration names for webhooks would we expect to see in a cluster with many webhooks? This seems like an unbounded dimension, but I wanted to gauge what sorts of different values we would expect to see practically. 

For instance, webhooks are theoretically unbounded but practically we wouldn't expect to see more a hundred different webhooks in a cluster, so ends up bounded in practice.

Reply:
For context, we have had cardinality issues with metrics in the past; this tends to manifest as memory leaks.

Reply:
@logicalhan Yes, in general, creating webhook with the same name is a user-determined behavior, it depends on whether the user already has that expectation when the same metric occurs.

Reply:
@logicalhan Have any suggestions with the issue #79832 ?

Reply:
My question still stands.. practically what sort of upper bound would we expect for configuration names for webhooks? 

/cc @sttts